service/history/queues/executable.go

```diff
@@ -135,6 +137,7 @@ func NewExecutable(
 	priorityAssigner PriorityAssigner,
 	timeSource clock.TimeSource,
 	namespaceRegistry namespace.Registry,
+	clusterMetadata cluster.Metadata,
```

Comment:
could it pass in the current cluster name?

Reply:
yeah that works too. Considering the # of task executable we are creating, I was thinking storing a reference can consume less memory than a string. But I understand it won't make too much difference, so I don't have strong opinion on either approach.